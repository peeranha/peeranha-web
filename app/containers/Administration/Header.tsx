import { useTranslation } from 'react-i18next';
import React from 'react';

import { MediumIconStyled } from 'components/Icon/MediumIcon';
import Wrapper from 'components/Header/Simple';
import Icon from 'components/Icon';
import H3 from 'components/H3';

import AddRoleForm from 'containers/Administration/AddRoleForm';
import { Moderator } from 'containers/Administration/types';

import AddModeratorButton from 'containers/Administration/AddModeratorButton';

import { singleCommunityColors } from 'utils/communityManagement';
import { BORDER_PRIMARY, ICON_TRASPARENT_BLUE } from 'style-constants';
import { Administration } from 'icons/index';
import { css } from '@emotion/react';

const colors = singleCommunityColors();

const customColor = colors.linkColor || BORDER_PRIMARY;

type HeaderProps = {
  locale: string;
  single?: number;
  moderators: Array<Moderator>;
  addRole: (userAddress: string, role: number, communityId: number) => void;
  addRoleLoading: boolean;
};

export const Header: React.FC<HeaderProps> = ({
  locale,
  single,
  moderators,
  addRole,
  addRoleLoading,
}): JSX.Element => {
  const { t } = useTranslation();

  return (
    <Wrapper className="mb-to-sm-0 mb-from-sm-3">
      <H3>
        <div
          css={css`
            .fill {
              fill: ${customColor};
            }
            .stroke {
              stroke: ${customColor};
            }

            .semitransparent {
              fill: ${colors.transparentIconColor || ICON_TRASPARENT_BLUE};
            }
          `}
        >
          <MediumIconStyled>
            <Administration />
          </MediumIconStyled>
        </div>
        {t('common.administration')}
      </H3>
      <AddRoleForm
        locale={locale}
        single={single}
        addRole={addRole}
        moderators={moderators}
        Button={AddModeratorButton}
        addRoleLoading={addRoleLoading}
      />
    </Wrapper>
  );
};

export default React.memo(Header);
