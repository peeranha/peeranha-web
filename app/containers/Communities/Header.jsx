import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import * as routes from 'routes-config';

import commonMessages from 'common-messages';

import createCommunityIcon from 'images/createCommunity.svg?inline';
import addIcon from 'images/add.svg?external';

import Icon from 'components/Icon';
import TransparentButton from 'components/Button/Contained/Transparent';
import SubHeaderWrapper from 'components/Header/Complex';
import NavigationButton from 'components/Button/Contained/Navigation';
import A from 'components/A';

import messages from './messages';
import languages from './languagesOptions';
import { GO_TO_CREATE_COMMUNITY_SCREEN_BUTTON_ID } from './constants';

const suggestedCommunitiesRoute = routes.suggestedCommunities();
const communitiesRoute = routes.communities();

const Header = ({
  goToCreateCommunityScreen,
  SubHeader,
  changeSorting,
  sorting,
  communitiesNumber,
  setLang,
  language,
}) => {
  const path = window.location.pathname + window.location.hash;

  return (
    <div className="mb-to-sm-0 mb-from-sm-3">
      <SubHeaderWrapper position="top">
        <div>
          <A to={communitiesRoute} href={communitiesRoute}>
            <NavigationButton isLink={path !== communitiesRoute}>
              <FormattedMessage {...commonMessages.communities} />
            </NavigationButton>
          </A>

          <A to={suggestedCommunitiesRoute} href={suggestedCommunitiesRoute}>
            <NavigationButton isLink={path !== suggestedCommunitiesRoute}>
              <FormattedMessage {...messages.voting} />
            </NavigationButton>
          </A>
        </div>

        <div className="right-panel">
          <TransparentButton
            id={`${GO_TO_CREATE_COMMUNITY_SCREEN_BUTTON_ID}_header`}
            onClick={goToCreateCommunityScreen}
            className="d-flex align-items-center"
          >
            <img
              className="d-none d-sm-inline-block"
              src={createCommunityIcon}
              alt="icon"
            />

            <Icon
              className="d-inline-block d-sm-none"
              icon={addIcon}
              width="14"
            />

            <span className="ml-1">
              <FormattedMessage {...messages.suggestCommunity} />
            </span>
          </TransparentButton>
        </div>
      </SubHeaderWrapper>

      <SubHeader
        changeSorting={changeSorting}
        sorting={sorting}
        communitiesNumber={communitiesNumber}
        setLang={setLang}
        language={language}
        languages={languages}
      />
    </div>
  );
};

Header.propTypes = {
  goToCreateCommunityScreen: PropTypes.func,
  SubHeader: PropTypes.any,
  changeSorting: PropTypes.func,
  sorting: PropTypes.object,
  communitiesNumber: PropTypes.number,
  setLang: PropTypes.func,
  language: PropTypes.object,
};

export default Header;
