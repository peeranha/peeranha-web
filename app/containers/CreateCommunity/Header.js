import React from 'react';
import { FormattedMessage } from 'react-intl';

import * as routes from 'routes-config';
import commonMessages from 'common-messages';
import { TEXT_PRIMARY } from 'style-constants';

import createCommunityHeader from 'images/communitiesHeader.svg?inline';
import closeIcon from 'images/closeCircle.svg?inline';

import A from 'components/A';
import H3 from 'components/H3';
import Span from 'components/Span';
import BaseRounded from 'components/Base/BaseRounded';
import { MediumImageStyled } from 'components/Img/MediumImage';

import messages from './messages';

export const Header = /* istanbul ignore next */ () => (
  <BaseRounded className="d-flex align-items-center justify-content-between mb-3">
    <H3 className="d-flex align-items-end">
      <MediumImageStyled src={createCommunityHeader} alt="questionIcon" />
      <FormattedMessage {...messages.newCommunity} />
    </H3>

    <A
      to={routes.communities()}
      href={routes.communities()}
      className="d-inline-flex align-items-center"
    >
      <img className="mr-1" src={closeIcon} alt="x" />
      <Span color={TEXT_PRIMARY}>
        <FormattedMessage {...commonMessages.close} />
      </Span>
    </A>
  </BaseRounded>
);

export default React.memo(Header);
