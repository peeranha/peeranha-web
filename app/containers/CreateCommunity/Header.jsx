import React from 'react';
import { FormattedMessage } from 'react-intl';

import * as routes from 'routes-config';
import commonMessages from 'common-messages';
import { TEXT_PRIMARY } from 'style-constants';

import createCommunityHeader from 'images/communitiesHeader.svg?inline';
import closeIcon from 'images/closeCircle.svg?inline';

import A from 'components/A';
import H3 from 'components/H3';
import Span from 'components/Span';
import Wrapper from 'components/Header/Simple';
import { MediumImageStyled } from 'components/Img/MediumImage';

import messages from './messages';

export const Header = () => (
  <Wrapper className="mb-to-sm-0 mb-from-sm-3">
    <H3>
      <MediumImageStyled src={createCommunityHeader} alt="create-community" />
      <FormattedMessage {...messages.newCommunity} />
    </H3>

    <div className="right-panel">
      <A to={routes.communities()}>
        <button>
          <img className="mr-1" src={closeIcon} alt="x" />
          <Span color={TEXT_PRIMARY}>
            <FormattedMessage {...commonMessages.close} />
          </Span>
        </button>
      </A>
    </div>
  </Wrapper>
);

export default React.memo(Header);
