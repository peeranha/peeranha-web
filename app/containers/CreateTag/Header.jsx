import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import * as routes from 'routes-config';
import { TEXT_PRIMARY, BORDER_PRIMARY } from 'style-constants';

import suggestTagIcon from 'images/tagsHeaderIcon.svg?external';
import closeIcon from 'images/closeCircle.svg?external';

import A from 'components/A';
import H3 from 'components/H3';
import Span from 'components/Span';
import Wrapper, { WrapperRightPanel } from 'components/Header/Simple';
import Icon from 'components/Icon';
import { IconMd } from 'components/Icon/IconWithSizes';
import { MediumIconStyled } from 'components/Icon/MediumIcon';

import { singleCommunityColors } from 'utils/communityManagement';

const colors = singleCommunityColors();

export const Header = ({
  title,
  closeRedirectPage,
  closeButtonAction = null,
}) => {
  const { t } = useTranslation();

  return (
    <Wrapper className="mb-to-sm-0 mb-from-sm-3">
      <H3>
        <MediumIconStyled>
          <Icon
            icon={suggestTagIcon}
            width="43"
            fill={colors.btnColor || BORDER_PRIMARY}
            isColorImportant
          />
        </MediumIconStyled>
        {title || t('tags.newTag')}
      </H3>

      <WrapperRightPanel className="right-panel">
        <A
          to={closeRedirectPage || routes.communities()}
          onClick={closeButtonAction}
        >
          <button>
            <IconMd
              className="mr-1"
              icon={closeIcon}
              fill={colors.btnColor || BORDER_PRIMARY}
              isColorImportant
            />
            <Span
              color={colors.btnColor || TEXT_PRIMARY}
              className="button-label"
            >
              {t('common.close')}
            </Span>
          </button>
        </A>
      </WrapperRightPanel>
    </Wrapper>
  );
};

Header.propTypes = {
  title: PropTypes.element,
  closeRedirectPage: PropTypes.string,
  closeButtonAction: PropTypes.func,
};

export default React.memo(Header);
