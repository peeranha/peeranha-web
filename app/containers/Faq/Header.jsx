import React from 'react';
import * as routes from 'routes-config';
import { FormattedMessage } from 'react-intl';

import commonMessages from 'common-messages';

import { TEXT_PRIMARY } from 'style-constants';
import faqPageHeader from 'images/faqPageHeader.svg?inline';
import infoIcon from 'images/icon-information.svg?inline';

import BaseRounded from 'components/Base/BaseRounded';
import { MediumImageStyled } from 'components/Img/MediumImage';
import H3 from 'components/H3';
import Span from 'components/Span';
import A from 'components/A';

import messages from './messages';

const Header = () => (
  <BaseRounded className="d-flex align-items-center justify-content-between mb-3">
    <H3 className="d-flex align-items-end">
      <MediumImageStyled src={faqPageHeader} alt="faqPageHeader" />
      <FormattedMessage {...messages.faq} />
    </H3>

    <A to={routes.support()} className="d-inline-flex align-items-center">
      <img className="mr-1" src={infoIcon} alt="x" />
      <Span color={TEXT_PRIMARY}>
        <FormattedMessage {...commonMessages.support} />
      </Span>
    </A>
  </BaseRounded>
);

export default React.memo(Header);
