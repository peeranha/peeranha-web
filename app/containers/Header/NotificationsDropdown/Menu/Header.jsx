import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';

import messages from 'common-messages';

import { BORDER_SECONDARY_LIGHT, TEXT_SECONDARY } from 'style-constants';

import Span from 'components/Span';

import closeIcon from 'images/closeGray.svg?inline';
import WidthCentered from '../../../../components/LoadingIndicator/WidthCentered';

const Container = styled.div`
  display: flex;
  width: 100%;
  padding: 16px;
  justify-content: space-between;
  border-bottom: 1px solid ${BORDER_SECONDARY_LIGHT};

  > div span:nth-child(1) {
    margin-right: 10px;
    font-weight: 600;
    font-size: 18px;
  }
`;

const Header = ({ notificationsNumber, onClose }) => (
  <Container>
    <div>
      <FormattedMessage {...messages.notifications} />
      <Span color={TEXT_SECONDARY}>{notificationsNumber}</Span>
    </div>
    <button onClick={onClose}>
      <img src={closeIcon} width="15" alt="close_icon" />
    </button>
  </Container>
);

Header.propTypes = {
  onClose: PropTypes.func,
  notificationsNumber: PropTypes.number.isRequired,
};

export default Header;
