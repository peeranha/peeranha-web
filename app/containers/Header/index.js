/**
 *
 * Header
 *
 */

import React from 'react';
import $ from 'jquery';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose, bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';

import {
  makeSelectAccount,
  makeSelectProfileInfo,
} from 'containers/AccountProvider/selectors';

import { makeSelectLocation } from 'containers/App/selectors';

import { showLoginModal } from 'containers/Login/actions';

import { LEFT_MENU_ID } from 'containers/LeftMenu/constants';

import View from './View';
import { HEADER_ID } from './constants';

export class Header extends React.PureComponent {
  componentDidMount() /* istanbul ignore next */ {
    this.animate();
  }

  componentDidUpdate() {
    $(`#${HEADER_ID}`).removeClass('sticky');
    $(`#${LEFT_MENU_ID}`).removeClass('sticky');
  }

  animate = /* istanbul ignore next */ () => {
    let lastScrollTop = 0;

    window.addEventListener(
      'scroll',
      event => {
        const st = window.pageYOffset || document.documentElement.scrollTop;

        const { scrollY } = event.currentTarget;
        const { innerHeight } = window;

        if (scrollY > innerHeight) {
          if (st > lastScrollTop) {
            $(`#${HEADER_ID}`).addClass('sticky');
            $(`#${LEFT_MENU_ID}`).addClass('sticky');
          } else {
            $(`#${HEADER_ID}`).removeClass('sticky');
            $(`#${LEFT_MENU_ID}`).removeClass('sticky');
          }
        }

        lastScrollTop = st <= 0 ? 0 : st;
      },
      false,
    );
  };

  render() /* istanbul ignore next */ {
    const {
      account,
      profileInfo,
      showLoginModalDispatch,
      showMenu,
      expandLeftMenuNavigation,
    } = this.props;

    return (
      <View
        account={account}
        profileInfo={profileInfo}
        showLoginModalDispatch={showLoginModalDispatch}
        showMenu={showMenu}
        expandLeftMenuNavigation={expandLeftMenuNavigation}
      />
    );
  }
}

Header.propTypes = {
  showLoginModalDispatch: PropTypes.func,
  account: PropTypes.string,
  profileInfo: PropTypes.object,
  showMenu: PropTypes.func,
  expandLeftMenuNavigation: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  account: makeSelectAccount(),
  profileInfo: makeSelectProfileInfo(),
  location: makeSelectLocation(),
});

export function mapDispatchToProps(dispatch) /* istanbul ignore next */ {
  return {
    showLoginModalDispatch: bindActionCreators(showLoginModal, dispatch),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(Header);
