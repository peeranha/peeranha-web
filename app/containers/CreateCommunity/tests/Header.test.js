import { Header } from '../Header';

describe('Header', () => {
  it('test', () => {
    expect(Header()).toMatchSnapshot();
  });
});
