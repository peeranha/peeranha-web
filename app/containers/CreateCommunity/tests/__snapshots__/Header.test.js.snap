// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test 1`] = `
<Base
  className="d-flex align-items-center justify-content-between mb-3"
>
  <Span
    className="d-flex align-items-end"
  >
    <Img
      alt="questionIcon"
      src="IMAGE_MOCK"
    />
    <FormattedMessage
      id="app.containers.CreateCommunity.newCommunity"
      values={Object {}}
    />
  </Span>
  <A
    className="d-inline-flex align-items-center"
    href="/communities"
    to="/communities"
  >
    <img
      alt="x"
      className="mr-1"
      src="IMAGE_MOCK"
    />
    <Span
      color="#576fed"
    >
      <FormattedMessage
        id="app.common.close"
        values={Object {}}
      />
    </Span>
  </A>
</Base>
`;
